javascript/README.md

```diff
@@ -728,3 +737,544 @@ function sayHi(name) {
+
+> Why? Rest arguments are a real Array, and not merely Array-like as the `arguments` object is.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+function concatenateAll() {
+  const args = Array.prototype.slice.call(arguments);
+  return args.join('');
+}
+
+// Slow performance
+function concatenateAll() {
+  const args = Array.from(arguments);
+  return args.join('');
+}
+```
+
+🎉 Yep! 🎉
+
+```js
+function concatenateAll(...args) {
+  return args.join('');
+}
+```
+
+#### Resources
+
+- ESLint: [prefer-rest-params]
+
+### 6.4 Function Default Parameters
+
+Use [function default parameter syntax][Function Default Parameters] rather than mutating function arguments.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+function doThings(opts) {
+  // If opts is falsey it can introduce bugs.
+  opts = opts || {};
+  // ...
+}
+
+function doThings(opts) {
+  if (opts === undefined) {
+    opts = {};
+  }
+  // ...
+}
+```
+
+🎉 Yep! 🎉
+
+```js
+function doThings(opts = {}) {
+  // ...
+}
+```
+
+#### Resources
+
+- MDN: [Function Default Parameters]
+
+### 6.5 Function Default Parameter Side Effects
+
+Avoid side effects with [function default parameters][Function Default Parameters].
+
+> Why? They are confusing to reason about.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+let b = 1;
+
+// Eek!
+function count(a = b++) {
+  console.log(a);
+}
+
+count();  // 1
+count();  // 2
+count(3); // 3
+count();  // 3
+```
+
+### 6.6 Function Constructor
+
+Never use the `Function` constructor to create a new function.
+
+> Why? Creating a function in this way evaluates a string similarly to `eval()`, which [opens vulnerabilities](#53-eval).
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+var add = new Function('a', 'b', 'return a + b');
+
+var subtract = Function('a', 'b', 'return a - b');
+```
+
+🎉 Yep! 🎉
+
+```js
+var x = function (a, b) {
+    return a + b;
+};
+```
+
+#### Resources
+
+- ESLint: [no-new-func]
+
+### 6.7 Mutating Function Parameters
+
+Never mutate function parameters.
+
+> Why? Manipulating objects passed in as parameters can cause unwanted variable side effects in the original caller.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+function foo(bar) {
+    bar = 13;
+}
+
+function foo(bar) {
+    bar++;
+}
+```
+
+🎉 Yep! 🎉
+
+```js
+function foo(bar) {
+    var baz = bar;
+}
+```
+
+#### Resources
+
+- ESLint: [no-param-reassign]
+
+
+### 6.8 Spread Syntax for Variadic Functions
+
+Prefer the use of the [spread syntax operator `...`][Spread Syntax] to call variadic functions (a function that accepts a variable number of arguments).
+
+> Why? It’s cleaner, you don’t need to supply a context, and it's easier to compose `new` when compared to using `apply`.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const args = [1, 2, 3, 4];
+Math.max.apply(Math, args);
+
+new (Function.prototype.bind.apply(Date, [null, 2016, 8, 5]));
+```
+
+🎉 Yep! 🎉
+
+```js
+const args = [1, 2, 3, 4];
+Math.max(...args);
+
+new Date(...[2016, 8, 5]);
+```
+
+#### Resources
+
+- ESLint: [prefer-spread]
```

Comment:
I think you should be linking to the built-in ESLint [`prefer-spread`](https://eslint.org/docs/rules/prefer-spread) rule here rather than the [`unicorn/prefer-spread`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/prefer-spread.md) rule. The `unicorn` rule enforces using `[...asdf]` instead of `Array.from(asdf)`, and the built-in rule enforces using `...` instead of `Function.prototype.apply`